### PR TITLE
kaslr aware address to symbol lookup

### PR DESCRIPTION
**Pull request by scorpionzezz**
original idea: https://github.com/libvmi/libvmi/pull/448

### CODE REVIEW

File: libvmi/os/linux/core.c [hunk 1]
```diff
@@ -360,6 +360,7 @@ status_t linux_init(vmi_instance_t vmi) {
     os_interface->os_ksym2v = linux_symbol_to_address;
     os_interface->os_usym2rva = NULL;
     os_interface->os_v2sym = linux_system_map_address_to_symbol;
```
Comment (tklengyel) [inline libvmi/os/linux/core.c]: This should be set to NULL now.

File: libvmi/os/linux/symbols.c [hunk 2]
```diff
@@ -210,6 +210,17 @@ char* linux_system_map_address_to_symbol(
     return NULL;
 }
 
+char* linux_system_map_address_to_symbol_kaslr(
```
Comment (tklengyel) [inline libvmi/os/linux/symbols.c]: Do we need a separate function for this? I think the kaslr offset subtraction could just be moved into linux_system_map_address_to_symbol directly.

File: libvmi/libvmi.h [hunk 3]
```diff
@@ -795,6 +795,11 @@ const char* vmi_translate_v2sym(
     const access_context_t *ctx,
     addr_t rva);
 
+const char* vmi_translate_v2ksym(
```
Comment (tklengyel) [inline libvmi/libvmi.h]: Missing function description comment.

File: libvmi/libvmi.h [hunk 4]
```diff
@@ -795,6 +795,11 @@ const char* vmi_translate_v2sym(
     const access_context_t *ctx,
     addr_t rva);
```
Comment (tklengyel) [inline libvmi/libvmi.h]: v2sym description should be updated saying Linux is not supported.

**Comment by scorpionzezz** [comment]
I changed it, I hope I get what you mean :)

File: libvmi/libvmi.h [hunk 5]
```diff
@@ -781,7 +781,7 @@ addr_t vmi_translate_sym2v(
 /**
  * Performs the translation from an RVA to a symbol
  * On Windows this function walks the PE export table.
- * The Linux translation works currently only for kernel symbols (pid=0 & base_vaddr=0).
+ * Not supported in Linux
  * Only the first matching symbol of System.map is returned.
```
Comment (tklengyel) [inline libvmi/libvmi.h]: This line is not required either.

File: libvmi/libvmi.h [hunk 6]
```diff
@@ -796,6 +796,21 @@ const char* vmi_translate_v2sym(
     addr_t rva);
 
 /**
+ * Performs the translation from an RVA to a symbol for Linux with KASLR offset
```
Comment (tklengyel) [inline libvmi/libvmi.h]: So this is not an RVA. This is an actual virtual address, so it should be called as such. We should also say this is not supported for Windows at the moment.

Comment (scorpionzezz) [inline libvmi/libvmi.h]: so maybe we should change it into VA instead or RVA ? :)

Comment (tklengyel) [inline libvmi/libvmi.h]: Yes, both in the comment and the variable name too.

**Comment by scorpionzezz** [comment]
changed from RVA to VA :)

**Comment by tklengyel** [comment]
This is ok to test.

File: libvmi/libvmi.h [hunk 7]
```diff
@@ -796,6 +795,22 @@ const char* vmi_translate_v2sym(
     addr_t rva);
 
 /**
+ * Performs the translation from a VA to a symbol for Linux with KASLR offset
+ * Windows is not supported at this moment
+ * Only the first matching symbol of System.map is returned.
+ * ELF Headers are not supported.
```
Comment (tklengyel) [inline libvmi/libvmi.h]: The comment about ELF headers is not relevant here.

**Comment by tklengyel** [comment]
Retest this please.

**Comment by tklengyel** [comment]
Looks good to me. Needs another +1 for the API addition.

@smaresca @awsaba @bdpayne

**Pull request merged**